src/output-json.c

```diff
@@ -767,6 +767,20 @@ static int CreateJSONEther(
             JSONFormatAndAddMACAddr(js, "src_mac", src, false);
             JSONFormatAndAddMACAddr(js, "dest_mac", dst, false);
             jb_close(js);
+        } else if (f != NULL) {
+            /* when pseudopackets do not have associated ethernet metadata,
+               use the first set of mac addresses stored with their flow */
+            MacSet *ms = FlowGetStorageById(f, MacSetGetFlowStorageID());
+            if (ms != NULL && MacSetSize(ms) > 0) {
+                uint8_t *src = MacSetGetFirst(ms, MAC_SET_SRC);
+                uint8_t *dst = MacSetGetFirst(ms, MAC_SET_DST);
+                if (dst != 0 || src != 0) {
```

Comment:
for pointers, please use `NULL`.

Also this looks a bit odd. We can continue if one of the pointers is NULL, is that intended?

Reply:
I think this scan-build warning is related
```
output-json.c:705:21: warning: Array access (from variable 'val') results in a null pointer dereference [core.NullDereference]
  705 |                     val[0], val[1], val[2], val[3], val[4], val[5]);
      |                     ^~~~~~
1 warning generated.
```